android/src/main/java/com/balthazargronon/RCTZeroconf/ZeroconfModule.java

```diff
@@ -49,62 +53,22 @@ public String getName() {
 
     @ReactMethod
     public void scan(String type, String protocol, String domain) {
-        if (mNsdManager == null) {
-            mNsdManager = (NsdManager) getReactApplicationContext().getSystemService(Context.NSD_SERVICE);
-        }
-
-        this.stop();
-
-        mDiscoveryListener = new NsdManager.DiscoveryListener() {
-            @Override
-            public void onStartDiscoveryFailed(String serviceType, int errorCode) {
-                String error = "Starting service discovery failed with code: " + errorCode;
-                sendEvent(getReactApplicationContext(), EVENT_ERROR, error);
-            }
-
-            @Override
-            public void onStopDiscoveryFailed(String serviceType, int errorCode) {
-                String error = "Stopping service discovery failed with code: " + errorCode;
-                sendEvent(getReactApplicationContext(), EVENT_ERROR, error);
-            }
-
-            @Override
-            public void onDiscoveryStarted(String serviceType) {
-                sendEvent(getReactApplicationContext(), EVENT_START, null);
-            }
-
-            @Override
-            public void onDiscoveryStopped(String serviceType) {
-                sendEvent(getReactApplicationContext(), EVENT_STOP, null);
-            }
-
-            @Override
-            public void onServiceFound(NsdServiceInfo serviceInfo) {
-                WritableMap service = new WritableNativeMap();
-                service.putString(KEY_SERVICE_NAME, serviceInfo.getServiceName());
-
-                sendEvent(getReactApplicationContext(), EVENT_FOUND, service);
-                mNsdManager.resolveService(serviceInfo, new ZeroResolveListener());
-            }
+        String serviceType = String.format("_%s._%s.", type, protocol);
 
-            @Override
-            public void onServiceLost(NsdServiceInfo serviceInfo) {
-                WritableMap service = new WritableNativeMap();
-                service.putString(KEY_SERVICE_NAME, serviceInfo.getServiceName());
-                sendEvent(getReactApplicationContext(), EVENT_REMOVE, service);
-            }
-        };
+        if (mDiscoverResolver == null) {
+            mDiscoverResolver = new MyDiscoverResolver(getReactApplicationContext(), serviceType, 1000);
```

Comment:
It's the debounceMillis, a delay time to wait until notify user from service drop (just to make sure it won't reappear that quick)